From 835a10f8541c7c4150098c82e097c4f606475cfa Mon Sep 17 00:00:00 2001
From: Tom de Vries <tdevries@suse.de>
Date: Mon, 20 Feb 2023 11:16:02 +0100
Subject: [PATCH] [gdb/testsuite] Add xfail case in
 gdb.python/py-record-btrace.exp

I came across:
...
gdb) PASS: gdb.python/py-record-btrace.exp: prepare record: stepi 100
python insn = r.instruction_history^M
warning: Non-contiguous trace at instruction 1 (offset = 0x3e10).^M
(gdb) FAIL: gdb.python/py-record-btrace.exp: prepare record: python insn = r.i\
nstruction_history
...

I'm assuming it's the same root cause as for the already present XFAIL.

Fix this by recognizing above warning in the xfail regexp.

Tested on x86_64-linux, although sofar I was not able to trigger the warning
again.

Approved-By: Markus T. Metzger <markus.t.metzger@intel.com>
---
 gdb/testsuite/gdb.python/py-record-btrace.exp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/gdb/testsuite/gdb.python/py-record-btrace.exp b/gdb/testsuite/gdb.python/py-record-btrace.exp
index ca8740bc967..a930d17264d 100644
--- a/gdb/testsuite/gdb.python/py-record-btrace.exp
+++ b/gdb/testsuite/gdb.python/py-record-btrace.exp
@@ -86,6 +86,11 @@ with_test_prefix "prepare record" {
 		  "warning: Decode error \\($nonl_re*\\) at instruction $decimal" \
 		  "\\(offset = $hex, pc = $hex\\):" \
 		  "$nonl_re*\\."]]
+    set xfail_re_2 \
+	[join \
+	     [list \
+		  "warning: Non-contiguous trace at instruction $decimal" \
+		  "\\(offset = $hex\\)\\."]]
 
     set got_xfail 0
     set cmd "python insn = r.instruction_history"
@@ -93,7 +98,7 @@ with_test_prefix "prepare record" {
 	-re "^[string_to_regexp $cmd]\r\n$::gdb_prompt $" {
 	    pass $gdb_test_name
 	}
-	-re -wrap "$xfail_re" {
+	-re -wrap "($xfail_re|$xfail_re_2)" {
 	    if { $have_xfail } {
 		xfail $gdb_test_name
 		set got_xfail 1

base-commit: f168a48adf97a36c88c65a075b42e6b7083063df
-- 
2.35.3

